### PR TITLE
Create rps.js

### PR DESCRIPTION
**Pull request by aakashrajput**
adding sparkychild rps script

### CODE REVIEW

File: chat-plugins/rps.js [hunk 1]
```diff
@@ -0,0 +1,321 @@
+/* * * * * * * * * * * *
+*  Rock/Paper/Scissors *
+*  by sparkychild      *
+* * * * * * * * * * * */
+'use strict';
+    
+    function rankLadder(title, type, array, prop, group) {
```
Comment (panpawn) [inline chat-plugins/rps.js:7]: indentation

File: chat-plugins/rps.js [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+		} else {
+			midColumn = '</td><td ' + tdStyle + '>';
+			tableRows += '<tr><td ' + tdStyle + '>' + (i + 1) + midColumn + array[i].name + midColumn + array[i][prop] + '</td></tr>';
+		}
+	}
+	return ladderTitle + tableTop + tableRows + tableBottom;
+}
+
+
+if (!Rooms.global.RPS) {
```
Comment (panpawn) [inline chat-plugins/rps.js:47]: um no

**Comment by CharizardtheFireMage** [comment]
You should really change from `Db('money')` to `Economy.readMoney`, or however the economy is run. (Sorry for trying to say that out panpawn and jd)

**Comment by aakashrajput** [comment]
no need to btw

File: chat-plugins/rps.js [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+		let gameResult = resultTable[this.p1choice.toLowerCase() + this.p2choice.toLowerCase()];
+		if (gameResult === "pp") {
+			//tie
+			this.p1.send("|pm|~Rock/Paper/Scissors Host|" + this.p1.userid + "|/html The game with " + this.p2.name + " was a tie! " + this.p2.name + " has chosen " + choiceNames[this.p2choice] + ".");
+			this.p2.send("|pm|~Rock/Paper/Scissors Host|" + this.p2.userid + "|/html The game with " + this.p1.name + " was a tie! " + this.p1.name + " has chosen " + choiceNames[this.p1choice] + ".");
+			if (this.gameType === "bucks") {
+				//return their 3 bucks each
+				Db('money').set(this.p1.userid, Db('money').get(this.p1.userid, 0) + 3);
+				Db('money').set(this.p2.userid, Db('money').get(this.p2.userid, 0) + 3);
+			}
```
Comment (CharizardtheFireMage) [inline chat-plugins/rps.js:161]: Pointing out that this script uses OriginDB, in which the economy is run through another source.

File: chat-plugins/rps.js [hunk 4]
```diff
@@ -0,0 +1,321 @@
...
+		this.p1.RPSgame = null;
+		this.p2.RPSgame = null;
+		delete Rooms.global.RPS.games[this.gameId];
+	}
+	parseWin(winner, loser, inactivity) {
+		winner.send("|pm|~Rock/Paper/Scissors Host|" + winner.userid + "|/html You have won the game against " + loser.name + "! " + (!inactivity ? loser.name + " has chosen " + choiceNames[(winner.userid === this.p1.userid ? this.p2choice : this.p1choice)] + "." : ""));
+		loser.send("|pm|~Rock/Paper/Scissors Host|" + loser.userid + "|/html You have lost the game against " + winner.name + "! " + (!inactivity ? winner.name + " has chosen " + choiceNames[(loser.userid === this.p1.userid ? this.p2choice : this.p1choice)] + "." : ""));
+		if (this.gameType === "bucks") {
+			//set but bucks
+			Db('money').set(winner.userid, Db('money').get(winner.userid, 0) + 6);
```
Comment (CharizardtheFireMage) [inline chat-plugins/rps.js:181]: Same for here.

File: chat-plugins/rps.js (hunk 2 above)
Comment (SilverTactic) [inline chat-plugins/rps.js:47]: just <code>global</code> would be preferable here, I guess

Comment (panpawn) [inline chat-plugins/rps.js:47]: better question is why does it need a global

**Comment by AlainDuhHero** [comment]
ye Wisp doesnt use phil's economy code it uses jd
which isnt based on origindb
So make it work with jd's economy code
@rambo199

**Comment by aakashrajput** [comment]
I need time guys.. i am working on some schul project you can see now these
days i am not so regular sry guys . Schul work

**Comment by panpawn** [comment]
This PR looks abandoned, closing for the time being

**Pull request closed without merging**